### PR TITLE
[Doc] Add table type comparison

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+<table>
+<thead>
+<tr><th> </th><th><strong>主键表 (Primary Key table)</strong></th><th><strong>明细表 (Duplicate Key table)</strong></th><th><strong>聚合表 (Aggregate table)</strong></th><th><strong>更新表 (Unique Key table)</strong></th></tr>
+</thead>
+<tr><td width= "200"><strong>Key 列和唯一约束</strong></td><td>主键（Primary Key）具有唯一约束和非空约束。</td><td>Duplicate Key 不具有唯一约束。</td><td>聚合键（Aggregate Key）具有唯一约束。</td><td>唯一键（Unique Key）唯一约束。</td></tr><tr><td><strong>Key 列和数据变更的关系（逻辑关系）</strong></td><td>基于主键对数据进行增删改操作。如果新数据的主键值与表中原数据的主键值相同，则存在唯一约束冲突，此时新数据会替代原数据。<br />与更新表相比，主键表增强了其底层存储引擎。已经可以取代更新表。</td><td>Duplicate Key 不具有唯一约束，因此如果新数据的 Duplicate Key 与表中原数据相同，则新旧数据都会存在表中。</td><td>基于聚合键对数据进行增删改操作。如果新数据与表中原数据存在唯一约束冲突，则会根据聚合键和 Value 列的聚合函数聚合新旧数据。</td><td>基于唯一键对数据进行增删改操作。如果新数据与表中原数据存在唯一约束冲突，则新数据会替代原数据。<br />更新表实际可以视为聚合函数为 replace 的聚合表。</td></tr><tr><td><strong>Key 列和排序键的关系</strong></td><td>自 3.0 起，两者解耦。</td><td colspan="3">两者耦合。</td></tr><tr><td><strong>Key 列和排序键支持的数据类型</strong></td><td>数值（包括整型、布尔）、字符串、时间日期。</td><td colspan="3">数值（包括整型、布尔、Decimal）、字符串、时间日期。</td></tr><tr><td><strong>Key 和分区/分桶列的关系</strong></td><td>分区列、分桶列必须在主键中。</td><td>无</td><td>分区列、分桶列必须在聚合键中。</td><td>分区列、分桶列必须在唯一键中。</td></tr>
```
Comment (amber-create) [inline docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md:9]: image
应该是漏了“具有”两个字。

Comment (amber-create) [inline docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md:9]: image
用逗号

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
.../tr><tr><td><strong>DML</strong> <strong><a href='../../sql-reference/sql-statements/data-manipulation/UPDATE.md'>UPDATE</a></strong> </td><td><ul><li>Key columns as filter conditions: Supported</li><li>Value columns as filter conditions: Supported </li></ul></td><td colspan="3">Not supported</td></tr><tr><td><strong>DML</strong> <strong><a href='../../sql-reference/sql-statements/data-manipulation/DELETE.md'>DELETE</a></strong></td><td><ul><li>Key columns as filter conditions: Supported</li><li>Value columns as filter conditions: Supported</li></ul></td><td><ul><li>Key columns as filter conditions: Supported</li><li>Value columns as filter conditions: Supported</li></ul>Note that only simple filter conditions based on key or value columns themselves are supported, such as =, &lt;, &gt;. Complex filter conditions, such as functions or subqueries, are not supported.</td><td colspan="2">Key columns as filter conditions: Supported. Note that only simple filter conditions based on key columns themselves are supported, such as =, &lt;, &gt;. Complex filter conditions, such as functions or subqueries, are not supported.Value columns as filter conditions: Not supported. </td></tr></tbody>
```
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:32]: image
Use lower case for these types of keys.

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
.... </td><td>Insert, delete, and update operations are performed based on the unique key.If the Unique Key value of a new data row is same as that of an existing data row in the table, the new data row will replace the existing data row.<br />The Unique Key table can be seen as an Aggregate table whose aggregate function is replace.</td></tr><tr><td><strong>Relationship between key columns and sort key</strong></td><td>Since v3.0, the key columns and sort key are decoupled. </td><td colspan="3">The key columns and sort key are coupled. </td></tr><tr><td><strong>Data types supported by key columns and sort key</strong>  </td><td>Numeric (including integers and BOOLEAN), string, and date (DATE and DATETIME). </td><td colspan="3">Numeric (including integers, BOOLEAN, and DECIMAL), string, and date (DATE and DATETIME).</td></tr><tr><td><strong>Relationship between key columns and partitioning/bucketing columns</strong> </td><td>Partitioning columns and bucketing columns must be in the primary key. </td><td>None </td><td>Partitioning columns and bucketing columns must be in the aggregate key. </td><td>Partitioning columns and bucketing columns must be in the unique key. </td></tr></tbody>
```
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:9]: image
Use lower case for these types of keys

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+</table>
+
+## Data types of key and value columns
+
+The key columns support the following data types: numeric (including integers, BOOLEAN, and DECIMAL), string, and date (DATE and DATETIME).
+:::note
```
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:15]: Leave an empty line above line 15

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+The key columns of the Primary Key table do not support the DECIMAL data type.
+:::
```
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:17]: Leave an empty line below line 17

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+</table>
+
+## Data types of key and value columns
+
+The key columns support the following data types: numeric (including integers, BOOLEAN, and DECIMAL), string, and date (DATE and DATETIME).
+:::note
+The key columns of the Primary Key table do not support the DECIMAL data type.
```
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:16]: the Primary Key table -> a Primary Key table

File: docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
...code>，或者列具有默认值，则无需导入全部列值。</td><td>默认必须导入全部列值。如果列具有默认值，则无需导入全部列值。</td><td>默认必须导入全部列值。不过，聚合表可以通过指定 Value 列的聚合函数为 REPLACE_IF_NOT_NULL 实现部分列更新，具体使用方式，请参见 <a href='../../sql-reference/sql-statements/data-definition/CREATE_TABLE.md#column_definition'>aggr_type</a>。并且如果列具有默认值，也无需导入全部列值。</td><td>默认必须导入全部列值。如果列具有默认值，则无需导入全部列值。</td></tr><tr><td><strong>DML</strong> <strong><a href='../../sql-reference/sql-statements/data-manipulation/INSERT.md'>INSERT</a></strong></td><td colspan="4">支持</td></tr><tr><td><strong>DML</strong><strong><a href='../../sql-reference/sql-statements/data-manipulation/UPDATE.md'>UPDATE</a></strong></td><td><ul><li>Key 列作为过滤条件：支持</li><li>Value 列作为过滤条件：支持</li></ul>   </td><td colspan="3">不支持</td></tr><tr><td><strong>DML</strong> <strong><a href='../../sql-reference/sql-statements/data-manipulation/DELETE.md'>DELETE</a></strong></td><td><ul><li>Key 列作为过滤条件：支持</li><li>Value 列作为过滤条件：支持</li></ul></td><td><ul><li>Key 列作为过滤条件：支持</li><li>Value 列作为过滤条件：支持注意，仅支持基于 Key 或 Value  列本身的简单过滤条件，如 =、&lt;、&gt;，不支持复杂条件，如函数、子查询。</li></ul></td><td  colspan="2"><ul><li>Key 列作为过滤条件： 支持。注意，仅支持基于 Key 列本身的简单过滤条件，如 =、&lt;、&gt;，不支持复杂条件，如函数、子查询。</li><li>Value 列作为过滤条件：不支持</li></ul></td></tr></tbody>
```
Comment (amber-create) [inline docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md:37]: image
这个“注意”后面应该有个句号。

Comment (amber-create) [inline docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md:37]: image
多余空格

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md (hunk 2 above)
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:32]: image
Add a period (.) inside the right-side parenthesis following each of these sentences.

Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:32]: image
"默认必须导入全部列值。" is missing.

Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:32]: image
Add a space preceding these two words "However".

Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:32]: image
The content in this cell should be made as a bulleted list.

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
...ong>Bitmap index/bloom filter index</strong></td><td><strong>Build indexes on key columns</strong></td><td colspan="4">Supported</td></tr><tr><td><strong>Build indexes on value columns</strong></td><td>Supported</td><td>Supported</td><td>Not supported</td><td>Not supported</td></tr><tr><td rowspan="2"><strong>Partitioning/bucketing</strong></td><td><strong>Expression partitioning/list partitioning</strong></td><td colspan="4">Supported</td></tr><tr><td><strong>Random bucketin</strong></td><td>Not supported</td><td>Supported from v3.1 onwards.</td><td>Not supported</td><td>Not supported</td></tr><tr><td  rowspan="2"><strong>Materialized views</strong></td><td ><strong>Asynchronous materialized views</strong></td><td colspan="4">Supported</td></tr><tr><td><strong>Synchronous materialized views</strong></td><td>Not supported</td><td>Supported</td><td>Supported</td><td>Supported</td></tr><tr><td rowspan="2"><strong>Other features</strong></td><td><strong>CTAS</strong></td><td>Supported</td><td>Supported</td><td>Not supported </td><td>Not supported </td></tr><tr><td><strong>backup &amp; restore</strong></td><td>Supported from v2.5 onwards.</td><td colspan="3">Supported</td></tr></tbody>
```
Comment (amber-create) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:42]: image
remove the periods (.)

**Review by amber-create — DISMISSED** [review]

File: docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+</table>
+
+## 和其他功能的兼容性
+
+<table>
+<thead>
+<tr><th colspan="2"></th><th><strong>主键表 (Primary Key table)</strong></th><th><strong>明细表 (Duplicate Key table)</strong></th><th><strong>聚合表 (Aggregate table)</strong></th><th><strong>更新表 (Unique Key table)</strong></th></tr>
+</thead>
+<tbody><tr><td rowspan="2"><strong>Bitmap</strong> <strong>索引/Bloom filer 索引</strong></td><td><strong>基于 Key 列构建索引</strong></td><td colspan="4">支持</td></tr><tr><td><strong>基于 Value 列构建索引</strong></td><td>支持</td><td>支持</td><td>不支持</td><td>不支持</td></tr><tr><td rowspan="2"><strong>分区/分桶</strong></td><td><strong>表达式分区/List 分区</strong></td><td colspan="4">支持</td></tr><tr><td><strong>随机分桶</strong></td><td>不支持</td><td>自 3.1 起，支持</td><td>不支持</td><td>不支持</td></tr><tr><td  rowspan="2"><strong>物化视图</strong></td><td colspan="4"><strong>异步物化视图</strong></td><td>支持</td></tr><tr><td><strong>同步物化视图</strong></td><td>不支持</td><td>支持</td><td>支持</td><td>支持</td></tr><tr><td rowspan="2"><strong>其他功能</strong></td><td><strong>CTAS</strong></td><td>支持</td><td>支持</td><td>不支持</td><td>不支持</td></tr><tr><td><strong>Backup &amp; restore</strong></td><td>自 2.5 起，支持</td><td colspan="3">支持</td></tr></tbody>
+</table>
```
Comment (evelynzhaojie) [inline docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md:47]: 这里是不是 tag 有问题
image

Comment (evelynzhaojie) [reply docs/zh/table_design/table_types/Comparison_of_abilities_of_various_tables.md:47]: 应该是异步物化视图是一个小单元格，支持合并了 4 种类型

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+</table>
```
Comment (evelynzhaojie) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:10]: operations are performed based on the unique key.If the unique key value   --- missing space between . and if

File: docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
```
Comment (evelynzhaojie) [inline docs/en/table_design/table_types/Comparison_of_abilities_of_various_tables.md:35]: loaded.However --- missing space

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Pull request merged**